host/lg/0021-keyboard-input-support-for-multi-windows.patch

```diff
@@ -0,0 +1,182 @@
+index ad81f14d..0bf0b276 100644
+--- a/client/include/vInputClient.h
++++ b/client/include/vInputClient.h
+@@ -1,11 +1,11 @@
+ #define XRES_MAX 600.00
+ #define YRES_MAX 960.00
+ #define LTRB_Y   292.00
+-typedef enum {PRESS, RELEASE, MOVE} action;
++typedef enum {PRESS, RELEASE, MOVE, KEYDOWN, KEYUP} action;
+ //typedef enum {ROTATION_0, ROTATION_1} rotation;
+ 
+ int initvInputClient(const char []);
+ void send_sync_event();
+ void send_event(int32_t type, int32_t code, int32_t value);
+-bool vinput_touch(action a,bool fullscreen, u_int32_t x, u_int32_t y, int rotation, u_int32_t dstRecth,u_int32_t dstRectw,u_int32_t dstRectx);
++bool vinput_touch(action a,bool fullscreen, u_int32_t x, u_int32_t y, int rotation, u_int32_t dstRecth,u_int32_t dstRectw,u_int32_t dstRectx,u_int32_t keys);
+ 
+diff --git a/client/src/main.c b/client/src/main.c
+index fd80333e..5bacfb53 100644
+--- a/client/src/main.c
++++ b/client/src/main.c
+@@ -98,7 +98,6 @@ static void updatePositionInfo()
+     state.dstRect.valid = true;
+     state.scaleX = (float)state.srcSize.y / (float)state.dstRect.h;
+     state.scaleY = (float)state.srcSize.x / (float)state.dstRect.w;
+-
+   }
+ 
+   state.lgrResize = true;
+@@ -685,7 +684,7 @@ void spiceClipboardRequest(const SpiceDataType type)
+ 
+ int eventFilter(void * userdata, SDL_Event * event)
+ {
+-  static bool serverMode   = false;
++ // static bool serverMode   = false;
+   static bool realignGuest = true;
+ 
+   switch(event->type)
+@@ -740,7 +739,7 @@ int eventFilter(void * userdata, SDL_Event * event)
+     case SDL_MOUSEMOTION:
+     {
+ #ifdef USE_INTELVTOUCH
+-      vinput_touch(MOVE,state.fullscreen, event->button.x, event->button.y, state.rotate, state.dstRect.h,state.dstRect.w,state.dstRect.x);
++      vinput_touch(MOVE,state.fullscreen, event->button.x, event->button.y, state.rotate, state.dstRect.h,state.dstRect.w,state.dstRect.x,event->key.keysym.scancode);
+       if (realignGuest)
+         realignGuest = false;
+ #else
+@@ -831,6 +830,12 @@ int eventFilter(void * userdata, SDL_Event * event)
+ 
+     case SDL_KEYDOWN:
+     {
++#ifdef USE_INTELVTOUCH
++    SDL_Scancode sc = event->key.keysym.scancode;
++    uint32_t scancode = mapScancode(sc);
++    vinput_touch(KEYDOWN,state.fullscreen, event->button.x, event->button.y, state.rotate, state.dstRect.h,state.dstRect.w,state.dstRect.x,scancode);
++#else
++
+       if (!spice_running)
+       {
+         if(params.useSpiceInput && !spice_ready())
+@@ -872,11 +877,18 @@ int eventFilter(void * userdata, SDL_Event * event)
+           break;
+         }
+       }
++#endif 
+       break;
+     }
++      
++   case SDL_KEYUP:
++   {
++#ifdef USE_INTELVTOUCH
++    SDL_Scancode sc = event->key.keysym.scancode;
++    uint32_t scancode = mapScancode(sc);
++    vinput_touch(KEYUP,state.fullscreen, event->button.x, event->button.y, state.rotate, state.dstRect.h,state.dstRect.w,state.dstRect.x,scancode);
++#else
+ 
+-    case SDL_KEYUP:
```

Comment:
why this case is removed?

Reply:
it is not removed.. it is there in line 85.